File: sql-data-partitioning/src/main/java/com/mageddo/sqldatapartitioning/dao/SkinPriceByDayDAO.java
package com.mageddo.sqldatapartitioning.dao;

import com.mageddo.sqldatapartitioning.entity.SkinPriceEntity;
import com.mageddo.sqldatapartitioning.enums.SkinPriceType;

import java.time.LocalDate;

public interface SkinPriceByDayDAO {
	void create(SkinPriceEntity skinPriceEntity);

	SkinPriceEntity find(LocalDate date, Long id);

	String createPartition(LocalDate date, SkinPriceType type);
}

File: sql-data-partitioning/src/main/java/com/mageddo/sqldatapartitioning/dao/SkinPriceByDayDAO.kt
package com.mageddo.sqldatapartitioning.dao

import com.mageddo.sqldatapartitioning.entity.SkinPriceEntity
import com.mageddo.sqldatapartitioning.enums.SkinPriceType

import java.time.LocalDate

interface SkinPriceByDayDAO {

	fun create(skinPriceEntity: SkinPriceEntity)

	fun find(date: LocalDate, id: Long?): SkinPriceEntity?

	fun createPartition(date: LocalDate, type: SkinPriceType): String
}
